hudi-common/src/main/java/org/apache/hudi/avro/MercifulJsonConverter.java

```diff
@@ -245,10 +245,14 @@ private static JsonToAvroFieldProcessor generateFixedTypeHandler() {
     return new JsonToAvroFieldProcessor() {
       @Override
       public Pair<Boolean, Object> convert(Object value, String name, Schema schema) {
-        byte[] src = value.toString().getBytes();
+        ArrayList<Integer> converval = (ArrayList<Integer>) value;
```

Comment:
Is this guaranteed to be list for a fixed decimal value ? Shouldn't this be scalar ?

Reply:
Current Fix.toString() will convert bytes to string, 
```
public String toString() { return Arrays.toString(bytes); } 
```
the result is like "decimal_val": [0, 0, 14, -63, -52], the generateFixedTypeHandler() will process it incorrect.

Reply:
@bvaradar Is it a bug?

Reply:
Thanks for the explanation. Makes sense. ObjectMapper uses List<Integer> to represent FixedType.  So, each entry in the list is within the range of [-128, 127]. right ?

Reply:
> Thanks for the explanation. Makes sense. ObjectMapper uses List to represent FixedType. So, each entry in the list is within the range of [-128, 127]. right ?

Yes.